mailgun-template-migration.py

```diff
@@ -1,75 +1,95 @@
+import http
 import os
+from argparse import ArgumentParser, Namespace
+from datetime import datetime
+from itertools import product
+from typing import List
+
+import requests
+from loguru import logger
+
+from models.Flags import Action, Env
+from models.TemplateModel import FullTemplate, BaseTemplate
 
 MG_BASE_URL = os.environ.get('MG_BASE_URL', 'https://api.mailgun.net/v3')
-MG_OLD_MAIL_DOMAIN = os.environ.get('MG_OLD_MAIL_DOMAIN', None)
-MG_NEW_MAIL_DOMAIN = os.environ.get('MG_NEW_MAIL_DOMAIN', None)
-MG_API_KEY = os.environ.get('MG_API_KEY', None)
-
-def get_template(name):
-    r = requests.get('{}/{}/templates/{}'.format(MG_BASE_URL, MG_OLD_MAIL_DOMAIN, name), auth=('api', MG_API_KEY), params={'active': 'yes'})
-    if r.status_code == 200:
-        return r.json()['template']
-    else:
-        return False
+MG_TEMPLATES_DOMAIN = os.environ.get('MG_TEMPLATES_DOMAIN')
+MG_LOCAL_MAIL_DOMAIN = os.environ.get('MG_LOCAL_MAIL_DOMAIN')
+MG_DEV_MAIL_DOMAIN = os.environ.get('MG_DEV_MAIL_DOMAIN')
+MG_PROD_MAIL_DOMAIN = os.environ.get('MG_PROD_MAIL_DOMAIN')
+MG_API_KEY = os.environ.get('MG_API_KEY')
 
-def create_template(name, description, template):
-    r = requests.post('{}/{}/templates'.format(MG_BASE_URL, MG_NEW_MAIL_DOMAIN), auth=('api', MG_API_KEY), data={'name': name, 'description': description, 'template': template})
-    if r.status_code == 200 and r.json()['message'] == 'template has been stored':
-        return r.json()['template']
-    else:
-        return False
 
-def delete_template(name):
-    r = requests.post('{}/{}/templates/{}'.format(MG_BASE_URL, MG_OLD_MAIL_DOMAIN, name), auth=('api', MG_API_KEY))
-    if r.status_code == 200 and r.json()['message'] == 'template has been deleted':
-        return True
-    else:
-        return False
-
-def get_all_templates():
-    templateNames = []
-    params = {}
-    print('Fetching all templates under domain {}'.format(MG_OLD_MAIL_DOMAIN))
-    while True:
-        r = requests.get('{}/{}/templates'.format(MG_BASE_URL, MG_OLD_MAIL_DOMAIN), auth=('api', MG_API_KEY), params=params)
-        if r.status_code == 200:
-            resp = r.json()
-            for item in resp['items']:
-                templateNames.append(item['name'])
-
-            if len(resp['items']) > 0:
-                params['page'] = 'next'
-                params['p'] = resp['items'][len(resp['items']) - 1]['name']
-            else:
-                break
+def read_arguments() -> Namespace:
+    parser = ArgumentParser(description='MailGun template migrator')
+    parser.add_argument('action', nargs='?', type=Action, default=Action.COPY,
+                        help='choose which action would you like to do. only *copy* supported')
+    parser.add_argument('env', type=Env, default=Env.DEV,
+                        help='choose which environment to copy templates from. local, dev, prod or ALL')
+    return parser.parse_args()
+
+
+def get_template(templates: List[BaseTemplate]) -> List[FullTemplate]:
```

Comment:
get_template_content (why not place the function in class?)

Reply:
done :)